testsuite/MDAnalysisTests/core/test_atomgroup.py

```diff
@@ -984,189 +938,197 @@ class TestUnwrapFlag(object):
-
-    def test_default(self, ref_noUnwrap):
+    def unwrap_group(self):
         u = UnWrapUniverse(is_triclinic=False)
-        group = u.atoms[31:39]  # molecules  11
-        # Changing masses for center_of_mass
+        group = u.atoms[31:39]
         group.masses = [100.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0]
+        return group
 
-        assert_almost_equal(group.center_of_geometry(), ref_noUnwrap['COG'], self.prec)
-        assert_almost_equal(group.center_of_mass(), ref_noUnwrap['COM'], self.prec)
-        assert_almost_equal(group.moment_of_inertia(), ref_noUnwrap['MOI'], self.prec)
-        assert_almost_equal(group.asphericity(), ref_noUnwrap['Asph'], self.prec)
-
-    def test_UnWrapFlag(self, ref_Unwrap):
-        u = UnWrapUniverse(is_triclinic=False)
-        group = u.atoms[31:39]  # molecules  11
-        # Changing masses for center_of_mass
-        group.masses = [100.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0]
-        assert_almost_equal(group.center_of_geometry(unwrap=True), ref_Unwrap['COG'], self.prec)
-        assert_almost_equal(group.center_of_mass(unwrap=True), ref_Unwrap['COM'], self.prec)
-        assert_almost_equal(group.moment_of_inertia(unwrap=True), ref_Unwrap['MOI'], self.prec)
-        assert_almost_equal(group.asphericity(unwrap=True), ref_Unwrap['Asph'], self.prec)
+    @pytest.mark.parametrize('unwrap, ref', ((True, ref_Unwrap_residues),
+                                             (False, ref_noUnwrap_residues)))
+    @pytest.mark.parametrize('method_name', ('center_of_geometry',
+                                             'center_of_mass',
+                                             'moment_of_inertia',
+                                             'asphericity'))
+    def test_residues(self, ag, unwrap, ref, method_name):
+        method = getattr(ag, method_name)
+        if unwrap:
+            result = method(compound='residues', unwrap=unwrap)
+        else:
+            # We test unwrap=False as the default behavior
+            result = method(compound='residues')
+        assert_almost_equal(result, ref[method_name], self.prec)
+
+    @pytest.mark.parametrize('unwrap, ref', ((True, ref_Unwrap),
+                                             (False, ref_noUnwrap)))
+    @pytest.mark.parametrize('method_name', ('center_of_geometry',
+                                             'center_of_mass',
+                                             'moment_of_inertia',
+                                             'asphericity'))
+    def test_group(self, unwrap_group, unwrap, ref, method_name):
+        method = getattr(unwrap_group, method_name)
+        if unwrap:
+            result = method(unwrap=unwrap)
+        else:
+            # We test unwrap=False as the default behavior
+            result = method()
+        assert_almost_equal(result, ref[method_name], self.prec)
 
 
 class TestPBCFlag(object):
 
     prec = 3
 
-    @pytest.fixture()
```

Comment:
I'd probably leave these as fixtures speaking stylistically. - It's a better way to express that they're constant across tests. But we're hardly consistent with this (see entire coordinates module tests...)

Reply:
(Good you brought this up, because I had wrongly set up the reference parametrization in the tests using this. Fixed that now and actually was able to simplify the entire class to a single parametrized test.)
The issue is that if I want to parametrize the reference to be used in a test, it cannot be a method (I mean, it can, but then the test must call the method to obtain the reference, which is quite ugly IMO). As regular attributes this just works cleanly.

I think an alternative, if we really want to keep the `@pytest.fixture` _and_ the parametrization simplification, is to parametrize the fixture to access the `request` argument and figure out whether it's being used in a test with `pbc=True` or `pbc=False` and generate the appropriate reference dict accordingly. I'm unsure about the neatness-to-overkill ratio in such an approach, so I'll follow what you suggest is best.